### ISSUE TITLE
Can't convert open Word, but can copy

### ISSUE THREAD

**Issue by ZhuangQu** [jgm/pandoc#10436]
**Explain the problem.**
Include the **exact command line** you used and **all inputs necessary to reproduce the issue**.  Please create as minimal an example as possible, to help the maintainers isolate the problem.  Explain the output you received and how it differs from what you expected.

**Pandoc version?**
What version of pandoc are you using, on what OS? (If it's not the latest release, please try with the latest release before reporting the issue. Note that many linux distributions have old versions of pandoc in their repositories.)

**Comment by jgm** [comment]
Pandoc shouldn't need higher permissions than copy.  Not sure what is going on here, because the error message is not very informative.

What are the permissions on A.docx?  (I assume you're on Windows; I don't know anything about how permissions work on Windows.)

Have you tried changing the permissions to see if it makes a difference for pandoc?

**Comment by ZhuangQu** [comment]
My OS version is Windows 11 23H2 and pandoc version is 3.5. The permission of A.docx is default (unchanged since creation). I did more experiments and found that the issue occurred while OneDrive running. Would you consider my suggestion? This will make pandoc compatible with OneDrive.

**Comment by jgm** [comment]
I really can't help until the issue is more precisely diagnosed.
Is the file in question on a cloud drive?

**Comment by ZhuangQu** [comment]
> I really can't help until the issue is more precisely diagnosed. Is the file in question on a cloud drive?

After reading your reply, I did another experiment and confirm your doubts. In general, this issue requires three conditions:
(1) A.docx is open.
(2) OneDrive is running.
(3) A.docx is in the OneDrive folder.

**Comment by jgm** [comment]
I don't think I can help further, because I don't have a Windows/OneDrive setup to test on.

**Comment by ZhuangQu** [comment]
Are there other developers/contributors can help

**Comment by ZhuangQu** [comment]
> Pandoc shouldn't need higher permissions than copy.

I wonder if pandoc will lock the file when reading it?

**Comment by jgm** [comment]
I can't really work on this unless someone gives me access to a machine with OneDrive that I could test on; it would also need to be possible to compile pandoc on that machine, so I can modify code and test.

**Comment by ZhuangQu** [comment]
You're right. We understand. Maybe there will be an opportunity to solve in the future.

**Comment by tarleb** [comment]
This here seems to be the most likely explanation that I could find: https://stackoverflow.com/q/44608214. Judging from that Q&A, the underlying issue is lazy I/O and Haskell's runtime system. From the answer (emphasis theirs):

> As far as I am aware, **there is no solution to this** in Data.ByteString.Lazy

**Comment by ZhuangQu** [comment]
Thank you for the explanation you found! This seems to be an unsolvable problem.

**Comment by jgm** [comment]
There are not many places in the code where we use IO with lazy ByteStrings.
Reading highlighting styles is one; reading data files is another -- but only when pandoc is not compiled with `embed_data_files` (and it is compiled with that flag in our releases).

Reading of source files uses strict bytestrings.

So, it's still puzzling to me where this comes from.
